src/content/blog/working-with-ddev-and-vite.md

```diff
@@ -0,0 +1,482 @@
+
+### 1. Expose the vite port
+
+Vite is installed within DDEV, but we can't access it from outside of the Docker container yet. We need to expose port `5173` of the DDEV project.
+
+Why? 
+
+If you use Vite on your localhost (without DDEV), the Vite development server would be accessible at `http://localhost:5173/`.
+
+Now we need it make it accessible via `https://test-vite.ddev.site:5173`.
+
+Fortunately exposing the port is very simple with DDEV's config option [web_extra_exposed_ports](https://ddev.readthedocs.io/en/latest/users/extend/customization-extendibility/#exposing-extra-ports-via-ddev-router). We add the following to `.ddev/config.yaml` file: 
+
+```yaml
+web_extra_exposed_ports:
+  - name: vite
+    container_port: 5173
+    http_port: 5172
+    https_port: 5173
+```
+
+This is our resulting `.ddev/config.yaml` file:
+
+```yaml
+name: test-vite
+type: php
+docroot: ""
+php_version: "8.1"
+webserver_type: nginx-fpm
+xdebug_enabled: false
+additional_hostnames: []
+additional_fqdns: []
+database:
+    type: mariadb
+    version: "10.4"
+use_dns_when_possible: true
+composer_version: "2"
+web_environment: []
+nodejs_version: "18"
+web_extra_exposed_ports:
+  - name: vite
+    container_port: 5173
+    http_port: 5172
+    https_port: 5173
+```
+
+A `ddev restart` is necessary after changing the config.yaml file.
+
+You can check the exposed ports with `ddev describe` after the restart.
+
+_Note: In other projects you might come across docker-compose-files in the .ddev/-folder which also take care of exposing the vite port. If you use `web_extra_exposed_ports`, you don't need these files._
+
+### 2. Adjust the Vite config
+
+The last step is to adjust the Vite config to let it know that it will run on `https://test-vite.ddev.site:5173`.
+
+This can be done easily by creating a `vite.config.js` file like this: 
+
+```js
+import { defineConfig } from 'vite'
+import path from 'path'
+
+const port = 5173;
+const origin = `${process.env.DDEV_PRIMARY_URL}:${port}`;
+
+// https://vitejs.dev/config/
+export default defineConfig({
+    // Add entrypoint
+    build: {
+        // our entry
+        rollupOptions: {
+          input: path.resolve(__dirname, 'src/main.js'),
+        },
+
+        // manifest
+        manifest: true
+      },
+
+    // Adjust Vites dev server for DDEV
+    // https://vitejs.dev/config/server-options.html
+    server: {
+        // respond to all network requests:
+        host: '0.0.0.0',
+        port: port,
+        strictPort: true, 
+        // Defines the origin of the generated asset URLs during development
+        origin: origin
+    },
+
+})
+```
+
+Technical explanation:
+
+- We need to define an entry file to let vite know where to start, this is done in `rollupOptions`. 
+- Also we need to tell vite to respond to all network request, not only the ones addressed to http://localhost:5173. This is done via `host: '0.0.0.0'`. You could also use `true`.
```

Comment:
I didn't understand "You could also use `true`" so suggest removing that, it's not clear what it's referring to.
```suggestion
- Also we need to tell vite to respond to all network request, not only the ones addressed to http://localhost:5173. This is done via `host: '0.0.0.0'`. 
```